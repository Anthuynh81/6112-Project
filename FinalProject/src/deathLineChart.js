import React, {useEffect, useState} from 'react';
import {Line} from 'react-chartjs-2';
import Axios from 'axios';

function DeathsLineChart() {
    const [data, setData] = useState([]);

    useEffect(() => {
        Axios.get("http://localhost:5000/USA").then((response) =>{
            setData(response.data)
            console.log(data)
        });
    }, []);

    const updatedColumn = (data) => data.map(x => x.Updated);
    const confirmedColumn = (data) => data.map(x => x.Deaths);

    const data1 = {
        labels: updatedColumn(data),
        datasets: [
            {
                label: 'Deaths Over Time',
                fill: false,
                lineTension: 0.1,
                backgroundColor: 'rgba(75,192,192,0.4)',
                borderColor: 'rgba(75,192,192,1)',
                borderCapStyle: 'butt',
                borderDash: [],
                borderDashOffset: 0.0,
                borderJoinStyle: 'miter',
                pointBorderColor: 'rgba(75,192,192,1)',
                pointBackgroundColor: '#fff',
                pointBorderWidth: 1,
                pointHoverRadius: 5,
                pointHoverBackgroundColor: 'rgba(75,192,192,1)',
                pointHoverBorderColor: 'rgba(220,220,220,1)',
                pointHoverBorderWidth: 2,
                pointRadius: 1,
                pointHitRadius: 10,
                data: confirmedColumn(data)
            }
        ]
    };

    return (
        <div>
            <h2>Line Chart</h2>
            <Line data={data1} />
        </div>
    );
}

export default DeathsLineChart;