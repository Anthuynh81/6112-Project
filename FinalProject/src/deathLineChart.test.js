import Axios from 'axios';
import renderer from "react-test-renderer";
import deathLineChart from "./deathLineChart";

const testDate = "03/14/2020"


async function getValue() {
    const response = await Axios.get("http://localhost:5000/USA")
    return response
}

test("Get request", async () => {
    const response = await getValue()
    //Success code for get request
    expect(response.status).toEqual(200)
})

test("Getting correct data", async () => {
    const response = await getValue()
    // Should be empty as it should have no state information
    expect(response.data[0].AdminRegion1).toBe("")
    // Should be empty as it should have no county information
    expect(response.data[0].AdminRegion2).toBe("\r")
})

test('returns the line chart when called', () => {
    const component = renderer.create(
        <deathLineChart />
    );
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})
