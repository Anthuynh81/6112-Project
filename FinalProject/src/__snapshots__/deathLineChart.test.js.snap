// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns the line chart when called 1`] = `<deathLineChart />`;
